src/model/compatibility-check/check-field.ts

```diff
@@ -35,3 +45,300 @@ export function checkField(fieldToCheck: Field, baselineField: Field, context: V
+
+    // superfluous inverseOf
+    if (fieldToCheck.inverseOf && !baselineField.inverseOf) {
+        context.addMessage(
+            ValidationMessage.compatibilityIssue(
+                `Relation "${
+                    baselineField.name
+                }" should be a forward relation, not an inverse relation${getRequiredBySuffix(
+                    baselineField,
+                )}.`,
+                fieldToCheck.inverseOfAstNode ?? fieldToCheck.astNode,
+            ),
+        );
+    }
+
+    // missing or wrong inverseOf
+    if (
+        baselineField.inverseOf &&
+        (!fieldToCheck.inverseOf || fieldToCheck.inverseOf.name !== baselineField.inverseOf.name)
+    ) {
+        context.addMessage(
+            ValidationMessage.compatibilityIssue(
+                `Relation "${
+                    baselineField.name
+                }" should be an inverse relation with ${expectedInverseOfDeclaration}${getRequiredBySuffix(
+                    baselineField,
+                )}.`,
+                fieldToCheck.inverseOfAstNode ??
+                    fieldToCheck.relationAstNode ??
+                    fieldToCheck.astNode,
+            ),
+        );
+    }
+
+    // wrong onDelete
+    if (fieldToCheck.relationDeleteAction !== baselineField.relationDeleteAction) {
+        // technically you can specify REMOVE_EDGES, but that's the default and nobody specifies that
+        const hint = expectedOnDeleteDeclaration
+            ? `specify ${expectedOnDeleteDeclaration}`
+            : `omit the "onDelete" argument`;
+        context.addMessage(
+            ValidationMessage.compatibilityIssue(
+                `Relation "${baselineField.name}" should ${hint}${getRequiredBySuffix(
+                    baselineField,
+                )}.`,
+                fieldToCheck.relationDeleteActionAstNode ??
+                    fieldToCheck.relationAstNode ??
+                    fieldToCheck.astNode,
+            ),
+        );
+    }
+}
+
+function checkCollectField(fieldToCheck: Field, baselineField: Field, context: ValidationContext) {
+    // superfluous @collect
+    if (fieldToCheck.isCollectField && !baselineField.isCollectField) {
+        context.addMessage(
+            ValidationMessage.compatibilityIssue(
+                `Field "${baselineField.declaringType.name}.${
+                    baselineField.name
+                }" should not be a collect field${getRequiredBySuffix(baselineField)}.`,
+                fieldToCheck.collectAstNode,
+            ),
+        );
+        return;
+    }
+
+    // no collect field on either side
+    if (!baselineField.isCollectField) {
+        return;
+    }
+
+    const expectedAggregateDeclaration = baselineField.aggregationOperator
+        ? `aggregate: ${baselineField.aggregationOperator.toString()}`
+        : '';
+
+    // missing @collect
+    if (!fieldToCheck.isCollectField) {
+        const expectedPathDeclaration = `path: "${baselineField.collectPath?.path ?? ''}"`;
```

Comment:
no, it would not be valid, but currently, you can call `checkCompatibility` given an invalid baselineProject. I think it's useful if validation etc. still somehow works with invalid projects and does not outright throw. (e.g. you could have an IDE that allows you to edit the project to check and the baseline model at the same time)